File: lib/NUnit.org/NUnit/2.5.9/samples/csharp/money/IMoney.cs
// ****************************************************************
// This is free software licensed under the NUnit license. You
// may obtain a copy of the license as well as information regarding
// copyright ownership at http://nunit.org/?p=license&r=2.4.
// ****************************************************************

namespace NUnit.Samples.Money 
{

	/// <summary>The common interface for simple Monies and MoneyBags.</summary>
	interface IMoney 
	{

		/// <summary>Adds a money to this money.</summary>
		IMoney Add(IMoney m);

		/// <summary>Adds a simple Money to this money. This is a helper method for
		/// implementing double dispatch.</summary>
		IMoney AddMoney(Money m);

		/// <summary>Adds a MoneyBag to this money. This is a helper method for
		/// implementing double dispatch.</summary>
		IMoney AddMoneyBag(MoneyBag s);

		/// <value>True if this money is zero.</value>
		bool IsZero { get; }

		/// <summary>Multiplies a money by the given factor.</summary>
		IMoney Multiply(int factor);

		/// <summary>Negates this money.</summary>
		IMoney Negate();

		/// <summary>Subtracts a money from this money.</summary>
		IMoney Subtract(IMoney m);
	}
}

File: lib/NUnit.org/NUnit/2.5.9/samples/vb/money/IMoney.vb
' ****************************************************************
' This is free software licensed under the NUnit license. You
' may obtain a copy of the license as well as information regarding
' copyright ownership at http://nunit.org/?p=license&r=2.4.
' ****************************************************************

Namespace NUnit.Samples

    'The common interface for simple Monies and MoneyBags.
    Public Interface IMoney

        'Adds a money to this money
        Function Add(ByVal m As IMoney) As IMoney

        'Adds a simple Money to this money. This is a helper method for
        'implementing double dispatch.
        Function AddMoney(ByVal m As Money) As IMoney

        'Adds a MoneyBag to this money. This is a helper method for
        'implementing double dispatch.
        Function AddMoneyBag(ByVal s As MoneyBag) As IMoney

        'True if this money is zero.
        ReadOnly Property IsZero() As Boolean

        'Multiplies a money by the given factor.
        Function Multiply(ByVal factor As Int32) As IMoney

        'Negates this money.
        Function Negate() As IMoney

        'Subtracts a money from this money.
        Function Subtract(ByVal m As IMoney) As IMoney

    End Interface

End Namespace
